app.py

```diff
@@ -241,6 +230,40 @@ def _get_contacts_data():
     return _contacts_data
 
 
+def _get_authorized():
+    """
+    Determine if the client is authorized
+
+    returns: True if authorized, False otherwise
+    """
+    if 'GRST_CRED_AURI_0' in request.environ:
+        # Ok, there may be a cert presented.  GRST_CRED_AURI_0 is the DN (probably).  Match that to something.
+        # Gridsite already made sure it matches something in the CA distribution
+
+        # HTTP unquote the DN:
+        client_dn = urllib.parse.unquote_plus(request.environ['GRST_CRED_AURI_0'])
+
+        # Get list of authorized DNs
+        authorized_dns = _get_dns()
+
+        # Authorized dns should be a set, or dict, that supports the "in"
+        if client_dn[3:] in authorized_dns: # "dn:" is at the beginning of the DN
```

Comment:
If we authorize based on DNs, we should verify the first three characters are `dn:`.